diff --git a/odoo/addons/base/models/ir_config_parameter.py b/odoo/addons/base/models/ir_config_parameter.py
--- a/odoo/addons/base/models/ir_config_parameter.py
+++ b/odoo/addons/base/models/ir_config_parameter.py
@@ -19,6 +19,7 @@ _default_parameters = {
     "database.secret": lambda: str(uuid.uuid4()),
     "database.uuid": lambda: str(uuid.uuid1()),
     "database.create_date": fields.Datetime.now,
+    "database.expiration_date": lambda: fields.Datetime.now().max.replace(microsecond=0),
     "web.base.url": lambda: "http://localhost:%s" % config.get('http_port'),
     "base.login_cooldown_after": lambda: 10,
     "base.login_cooldown_duration": lambda: 60,
